src/intrinsics/ecma262/Array.js

```diff
@@ -85,7 +86,12 @@ export default function (realm: Realm): NativeFunctionValue {
         // c. Let intLen be 1.
         intLen = 1;
       } else { // 7. Else,
-        len = len.throwIfNotConcreteNumber();
+        // len = recoverableError(() => len.throwIfNotConcreteNumber(), ()=>new NumberValue(realm, 0));
+        len = recoverableError(
```

Comment:
Based on discussion with Herman, the plan is to focus this change on introducing the mechanism needed for recovering from errors, while leveraging it in specific places where we want to support recovery will be done separately (and probably gradually).

Using the new mechanism in this location is intended to:
1. Serve as a sanity check + usage example
2. Allow for testing (I'm working on the tests and they'll be included in the next iteration of the review)